keanu-project/src/main/java/io/improbable/keanu/vertices/bool/nonprobabilistic/operators/BoolModelResultVertex.java

```diff
@@ -0,0 +1,42 @@
+package io.improbable.keanu.vertices.bool.nonprobabilistic.operators;
+
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.vertices.NonProbabilistic;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.VertexLabel;
+import io.improbable.keanu.vertices.bool.BoolVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.multiple.ModelVertex;
+
+/**
+ * A non-probabilistic boolean vertex whose value is extracted from an upstream model vertex.
+ */
+public class BoolModelResultVertex extends BoolVertex implements NonProbabilistic<BooleanTensor> {
+
+    private ModelVertex model;
+    private VertexLabel label;
+
+    public BoolModelResultVertex(ModelVertex model, VertexLabel label) {
+        this.model = model;
+        this.label = label;
+        setParents((Vertex) model);
+    }
+
+    @Override
+    public BooleanTensor getValue() {
+        if (!model.hasCalculated()) {
```

Comment:
Do we have an issue where we will only calculate once?  Can we just make "eval" on the model vertex itself do the work?

Reply:
Added a test for this - `canRunEvalOnTheOutputsToRecalculateTheModel`